### DISCUSSION TITLE
Stops after starting job (Dreame Z10 Pro )

### DISCUSSION THREAD

**Discussion by 13peter13** [Hypfer/Valetudo#1803] [Q&A / Support]
Hello!

After the last update today to Valetudo 2023.03.0 I have the Problem that short after the start of a cleaning job the robot just stops and restarts after a while. Before that it worked like a charm for over a year; FW is 1156

Debug log until the crash:

              Debug <<< cloud* { stamp: 1682180592 }
              2023-04-22T16:23:11.964Z Debug >>> cloud* { stamp: 1682180592 }
              2023-04-22T16:23:12.622Z Debug >>> cloud: {
                method: 'action',
                params: {
                  did: 520993851,
                  siid: 6,
                  aiid: 1,
                  in: [
                    {
                      piid: 2,
                      value: '{"frame_type":"I", "force_type": 1, "req_type": 1}'
                    }
                  ]
                },
                id: 60
              }
              2023-04-22T16:23:12.638Z Debug <<< cloud: {
                id: 60,
                result: {
                  did: '520993851',
                  siid: 6,
                  aiid: 1,
                  out: [
                    { piid: 1, value: '' },
                    { piid: 3, value: '' },
                    { piid: 5, value: 1682180592633 }
                  ],
                  code: 0
                },
                exe_time: 7
              }
              2023-04-22T16:23:13.088Z Debug
              FDS upload started with: {
                query: {
                  ts: '41109565933',
                  suffix: 'urls',
                  Expires: '1682181274',
                  index: '1',

then after the robot restarted I tried to send it to the base and it crashes again:

            2023-04-22T16:31:39.079Z Debug
            >>> cloud: {
              method: 'action',
              params: {
                did: 520993851,
                siid: 6,
                aiid: 1,
                in: [
                  {
                    piid: 2,
                    value: '{"frame_type":"I", "force_type": 1, "req_type": 1}'
                  }
                ]
              },
              id: 46
            }
            2023-04-22T16:31:39.091Z Debug
            <<< cloud: {
              id: 46,
              result: {
                did: '520993851',
                siid: 6,
                aiid: 1,
                out: [
                  { piid: 1, value: '' },
                  { piid: 3, value: '' },
                  { piid: 5, value: 1682181099085 }
                ],
                code: 0
              },
              exe_time: 3
            }
            2023-04-22T16:31:39.283Z Debug
            FDS upload started with: {
              query: {
                ts: '41961260184',
                suffix: 'urls',
                Expires: '1682181750',
                index: '0',
                method: '_sync.gen_tmp_presigned_url'
              },
              params: { filename: undefined }
            }
            2023-04-22T16:31:40.968Z Debug
            FDS upload started with: {
              query: {
                ts: '41961260184',
                suffix: 'urls',
                Expires: '1682181750',
                index: '1',
                method: '_sync.gen_tmp_presigned_url'
              },
              params: { filename: undefined }
            }
            2023-04-22T16:31:41.603Z Debug >>> cloud: {
              method: 'action',

I have tried to downgrade valetudo to previous versions without any success, also I have tried to reinstall the firmware with the dustbuilder image also with no success.

I have no idea how to proceed from here to get it working again ;(

Hopefully you have some ideas! If you need more information please let me know.

Thanks in advance!

**Comment by Hypfer** [comment]
Unfortunately it's not unlikely that you're facing a hardware defect that just happened to show up at a similar time as the update :(

The Z10 and L10 do occasionally do that. It seems to often be related to one or both of the RAM chips not being soldered correctly and thus loosing connection to the board. Come to think of it, we also have someone with an L10S Pro with similar symptoms.

Sometimes you can tell that that happened by the robot only reporting 256MB of RAM. Sometimes both chips fail, leading to it not booting at all anymore and sometimes it's a transient issue where the system behaves erratic with processes segfaulting at random times due to memory corruption.

**Reply by 13peter13** [reply]
So first of all thanks for your quick reply!

When I check in the system information the RAM adds up to 560,51 MiB 
From the software point of view is there some other way to verify that it's the RAM?

If there is a problem with the RAM chips not being soldered correctly do you think its possible to open the robot and re-solder it?
Is there any other way to repair the robot or is it dead? :(

**Reply by Hypfer** [reply]
> do you think its possible to open the robot and re-solder it

Only with the right equipment but that's something very advanced to do.
It's also something that should not be done as the first step when troubleshooting.

I was just mentioning that because if you still had warranty, it might've been a warranty case.

Either way, this needs more info. What does "just stops and restarts after a while" mean exactly?
The logs attached only show regular operation

**Reply by 13peter13** [reply]
Okay I'll try if it is a warranty case. If I have to send it away do I have to change something on the software before that?

So the case withe the robot is as follows. I give it a job and then it says "start selected room cleaning" drives out of the dock, turns and starts the vacuum and the brush. After about 2 seconds driving its just stops; the power button is still lit. At this point it is not reachable anymore. If I wait some minutes it restarts (makes the starting sound) or I can also force a start by first shutting it down (pressing the power button long) and starting it again.

**Reply by Hypfer** [reply]
Did you also use FW 1156 before this happened?

**Reply by 13peter13** [reply]
I am not quite sure.

Does the downgrade of the FW worke the same way as the upgrade? The I'll try that

**Reply by 13peter13** [reply]
So I successfully installed FW 1059 but still the same problem. As next step I downgraded Valetudo to 2023.02.0 but still the same

An interesting thing, when I start a new job the robot needs to position it self and that works fine, then it wants to proceed with the job and after some seconds it's dead again.

Any suggestions?

**Reply by Hypfer** [reply]
Sounds like the MCU firmware issues that the D9 has.
See here:
https://github.com/Hypfer/Valetudo/discussions/1556

**Reply by 13peter13** [reply]
So i have followed the instructions from #1556 unfortunately withe no success. I have also tried to install the MCU firmware multiple times as suggested. One time the robot drove further but stoppt at the border of the area to clean. Now its still the same behavior as before. The robot drives out of the dock, turns around and when it starts the vacuum it crashes again.

**Reply by dgiese** [reply]
connect to the other UART port (of the MCU) and read out what version number it gives you. Maybe the versions are still mismatched and thats why the update fails. Typically, we would let the robot sit for like 12 hours outside of the station until it shuts down internally. And then boot it again via the docking station.

**Reply by 13peter13** [reply]
Okay, I have started a job on the robot again and no wonder there it stops after some seconds and restarts after some minutes. 

If i got it right I should let it sit for about 12 hours and then try again. 

If no success connect to the UART port like described [here](https://valetudo.cloud/pages/installation/dreame.html). Will i see the MCU version after I start the connected robot or how do I proceed?

Thank you guys so much already, it's good to have people to help <3

**Reply by 13peter13** [reply]
So guys after some writing with the vendor he said I can send it back and he will forward it to the costumer support.

Before I send it should I change something on the software (upload the initial backup?) and if so how do I do that?
